fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/rdf/impl/TypeRdfContext.java

```diff
@@ -58,71 +61,36 @@ public TypeRdfContext(final FedoraResource resource,
 
         //include rdf:type for primaryType, mixins, and their supertypes
         concatRdfTypes();
+        if (resource() instanceof NonRdfSource) {
+            // gather versionability info from the parent
+            if (resource().getNode().getParent().isNodeType("mix:versionable")) {
```

Comment:
I'm not sure whether your suggestion is more maintainable or not, because we don't want to conflate description and versioning lifecycle. IOW, I'm looking at the parent _not_ because it is the description, but _specifically_ because it is the parent. If it wasn't the description and the description lived elsewhere (which we can easily imagine supporting) I would still be looking at the parent. I wouldn't go and trace through to the description. That's because it is the parent that controls the versionability, not the description. It just happens that for the moment, in this case, they are the same.